src/main/java/be/seeseemelk/mockbukkit/entity/PigMock.java

```
@@ -0,0 +1,73 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import com.google.common.base.Preconditions;
+import org.bukkit.Material;
+import org.bukkit.entity.Pig;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class PigMock extends AnimalsMock implements Pig
+{
+
+	private boolean hasSaddle = false;
+	private int boostTicks = 0;
+	private int currentBoostTicks = 0;
+
+	public PigMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+		boostTicks = ThreadLocalRandom.current().nextInt(841) + 140;
+	}
+
+	@Override
+	public boolean hasSaddle()
+	{
+		return this.hasSaddle;
+	}
+
+	@Override
+	public void setSaddle(boolean saddled)
+	{
+		this.hasSaddle = saddled;
+	}
+
+	@Override
+	public int getBoostTicks()
+	{
+		return this.boostTicks;
+	}
+
+	@Override
+	public void setBoostTicks(int ticks)
+	{
+		Preconditions.checkArgument(ticks >= 0, "ticks must be bigger than or equal to 0");
+		this.boostTicks = ticks;
+	}
+
+	@Override
+	public int getCurrentBoostTicks()
+	{
+		return this.currentBoostTicks;
+	}
+
+	@Override
+	public void setCurrentBoostTicks(int ticks)
+	{
+		if (!this.hasSaddle)
+		{
+			return;
```

Should this throw an exception instead of returning?

---

Closed since we should replicate Bukkit Behaviour (Reference Discord for Discussion)